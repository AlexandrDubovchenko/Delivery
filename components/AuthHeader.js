import React from 'react';
import { StyleSheet, View, Text } from 'react-native';
import { ExitButton } from './ExitButton';
import { TouchableOpacity } from 'react-native-gesture-handler';

export const AuthHeader = (props) => {
    const switchText = props.switch.substring(0, 4) + ' ' + props.switch.substring(4)
    return (
        <View style={styles.header}>
            <ExitButton navigation={props.navigation} />
            <TouchableOpacity onPress={() => props.navigation.replace(props.switch)}><Text style={[styles.button]}>{switchText}</Text></TouchableOpacity>
        </View>
    )
}

const styles = StyleSheet.create({
    button: {
        color: "#006400",
        marginTop: 25,
        fontSize: 20
    },
    header: {
        flex: .5,
        flexDirection: "row",
        justifyContent: "space-between"
    },
})