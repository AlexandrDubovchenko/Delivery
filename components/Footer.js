import React from 'react';
import { TouchableOpacity } from 'react-native-gesture-handler';
import { StyleSheet, View, Image } from 'react-native';
import * as RootNavigation from '../RootNavigation';

const Footer = (props) => {
 
    return (
        <View style={styles.footer}>
            <TouchableOpacity onPress={() => RootNavigation.navigate('HomeScreen')}>
                <Image style={styles.footer_icon} source={require('../assets/Footer/list.png')} />
            </TouchableOpacity>
            <TouchableOpacity onPress={() => RootNavigation.navigate('SearchScreen')}>
                <Image style={styles.footer_icon} source={require('../assets/Footer/search.png')} />
            </TouchableOpacity>
            <TouchableOpacity onPress={() => RootNavigation.navigate('ProfileScreen')}>
                <Image style={styles.footer_icon} source={require('../assets/Footer/person.png')} />
            </TouchableOpacity>
        </View>
    )
}

const styles = StyleSheet.create({
    footer: {
        flex: 0.1,
        flexDirection: "row",
        justifyContent: "space-around",
        alignItems: "center",
        backgroundColor: '#006400',
    },
    footer_icon: {
        height: 40,
        width: 40
    }
});

export default Footer