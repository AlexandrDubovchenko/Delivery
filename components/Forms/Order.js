/* eslint-disable global-require */
/* eslint-disable no-shadow */
import React, { useState } from 'react';
import {
  StyleSheet, View, Text, Image,
} from 'react-native';
import { TouchableOpacity } from 'react-native-gesture-handler';
import { connect } from 'react-redux';
import { reduxForm, Field } from 'redux-form';
import { showMessage } from 'react-native-flash-message';
import FormInput from './Controlers/Input';
import { required, telefoneNumber } from '../../utilities/validation';
import { orderAPI } from '../../api';
import { resetBasket } from '../../redux/reducers/basket-reducer';
import OrderButton from '../Button';

const styles = StyleSheet.create({
  form: {
    flex: 3,
  },
  totalPrice: {
    marginHorizontal: 15,
    marginTop: 10,
    fontSize: 20,
    marginBottom: 30,
  },
  mapButton: {
    flex: 1,
    position: 'absolute',
    right: 30,
    top: 7,
  },
  mapIcon: {
    width: 30,
    height: 40,
  },
});

export const OrderForm = ({
  handleSubmit, orderDishes, totalPrice, navigation, resetBasket, id,
}) => {
  const onSubmit = ({ name, address, telefone }) => {
    orderAPI.sendOrder({
      name, address, telefone, totalPrice, id, orderDishes,
    });
    showMessage({
      message: 'Ваш заказ отправлен, ожидайте звонка',
      type: 'success',
      duration: 9000,
      position: 'center',
    });
    navigation.navigate('HomeScreen');
    resetBasket();
  };

  const [address, setAddress] = useState('');
  return (
    <View category style={styles.form}>
      <Field placeholder="Имя" name="name" component={FormInput} validate={[required]} />
      <View>
        <Field placeholder="Aдрес" name="address" props={{ value: address }} component={FormInput} validate={[required]} />
        <View style={styles.mapButton}>
          <TouchableOpacity onPress={() => navigation.navigate('MapScreen', { setAddress })}>
            <Image style={styles.mapIcon} source={require('../../assets/Orders/map_icon.png')} />
          </TouchableOpacity>
        </View>
      </View>


      <Field placeholder="Телефон" name="telefone" component={FormInput} validate={[required, telefoneNumber]} />
      <Text style={styles.totalPrice}>
        Сумма заказа:
        {totalPrice}
        грн
      </Text>
      {orderDishes.length
        ? <OrderButton onSubmit={handleSubmit(onSubmit)} text="Отправить заказ" />
        : <Text style={styles.totalPrice}>Вы ничего не заказали</Text>}
    </View>
  );
};

const OrderReduxForm = reduxForm({ form: 'order' })(OrderForm);


const mapStateToProps = (state) => ({
  orderDishes: state.basket.basketList,
  totalPrice: state.basket.totalPrice,
  id: state.auth.id,
});
export default connect(mapStateToProps, { resetBasket })(OrderReduxForm);
