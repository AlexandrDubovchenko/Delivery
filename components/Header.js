/* eslint-disable global-require */
import React from 'react';
import { connect } from 'react-redux';
import {
  StyleSheet, View, Text, Image,
} from 'react-native';
import { TouchableOpacity } from 'react-native-gesture-handler';
import * as RootNavigation from '../RootNavigation';

const styles = StyleSheet.create({
  header: {
    flex: 0.1,
    backgroundColor: '#FFD700',
    flexDirection: 'row',
    justifyContent: 'space-around',
    alignItems: 'center',
  },
  buttonImg: {
    height: 40,
    width: 40,
  },
  basketButton: {
    flexDirection: 'row',
    alignItems: 'flex-end',
  },

});

// eslint-disable-next-line no-shadow
const Header = ({ isBasket, basketItemsNumber }) => (
  <View style={styles.header}>
    <TouchableOpacity onPress={() => RootNavigation.goBack()}>
      <Image style={styles.buttonImg} source={require('../assets/Header/back.png')} />
    </TouchableOpacity>

    {!isBasket
      ? (
        <TouchableOpacity onPress={() => RootNavigation.navigate('BasketScreen')}>
          <Image style={styles.buttonImg} source={require('../assets/Header/basket.png')} />
          {
            basketItemsNumber
              ? <Text>{basketItemsNumber}</Text> : null
          }
        </TouchableOpacity>
      )
      : (
        <TouchableOpacity style={styles.basketButton} onPress={() => RootNavigation.navigate('OrderScreen')}>
          <Text>Оформить заказ</Text>
        </TouchableOpacity>
      )}
  </View>
);

const mapStateToProps = (state) => ({
  basketItemsNumber: state.basket.basketList.length,
});

export default connect(mapStateToProps, null)(Header);
