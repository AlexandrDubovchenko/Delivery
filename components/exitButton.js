import React from 'react';
import { Image, StyleSheet } from 'react-native';
import { TouchableOpacity } from 'react-native-gesture-handler';

const styles = StyleSheet.create({
  img: {
    marginTop: 20,
    height: 50,
    width: 50,
  },
});

export const ExitButton = (props) => (
  <TouchableOpacity onPress={() => props.navigation.navigate('WelcomeScreen')}>
    <Image
      style={styles.img}
      source={{ uri: 'https://cdn.iconscout.com/icon/free/png-256/close-150-458082.png' }}
    />
  </TouchableOpacity>
);

export default ExitButton;
