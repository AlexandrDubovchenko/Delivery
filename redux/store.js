import { combineReducers, createStore, applyMiddleware } from "redux";
import authReducer from "./reducers/auth-reducer";
import { reducer as formReducer } from 'redux-form'
import thunkMiddleware from "redux-thunk";

const reducers = combineReducers({
    auth: authReducer,
    form: formReducer
})

const store = createStore(reducers, applyMiddleware(thunkMiddleware));

export default store