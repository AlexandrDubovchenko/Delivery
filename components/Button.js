import React from 'react';
import { TouchableOpacity } from 'react-native-gesture-handler';
import { StyleSheet } from 'react-native';
import { Text } from '@ui-kitten/components';

const styles = StyleSheet.create({
  orderButton: {
    alignItems: 'center',
    backgroundColor: '#FFD700',
    marginHorizontal: 15,
  },
});

const OrderButton = ({ onSubmit, text }) => (
  <TouchableOpacity
    style={styles.orderButton}
    onPress={onSubmit}
  >
    <Text category="h5">{text}</Text>
  </TouchableOpacity>
);

export default OrderButton;
