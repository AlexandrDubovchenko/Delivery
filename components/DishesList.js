import React from 'react';
import { StyleSheet, View, Image, } from 'react-native';
import { Card, List, Text } from '@ui-kitten/components';
import { TouchableOpacity } from 'react-native-gesture-handler';

export const DishesList = (props) => {

    const renderItemHeader = (headerProps, info) => (
        <View {...headerProps}>
            <Text category='h6'>
                {info.item.name}
            </Text>
        </View>
    );

    const renderItemFooter = (footerProps, info) => (
        <TouchableOpacity style={styles.orderButton} onPress={() => props.setBasketItem(info.item.name, info.item.img)}>
            <Text category='h6'>
                Заказать
        </Text>
        </TouchableOpacity>
    );



    const renderItem = (info) => (
        <Card
            style={styles.item}
            header={headerProps => renderItemHeader(headerProps, info)}
            footer={(footerProps) => renderItemFooter(footerProps, info)}
        >
            <Image style={styles.categoryImage} source={{ uri: info.item.img }} />
        </Card>
    );

    return (
        <List
            style={styles.container}
            contentContainerStyle={styles.contentContainer}
            data={props.data}
            renderItem={renderItem}
        />
    );
};

const styles = StyleSheet.create({
    contentContainer: {
        paddingHorizontal: 8,
        paddingVertical: 4,
    },
    item: {
        marginVertical: 4,

    },
    categoryImage: {
        height: 200
    },
    orderButton: {
        alignItems: "center",
        backgroundColor: "#FFD700"
    }
});