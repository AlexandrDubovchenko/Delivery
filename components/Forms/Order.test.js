import * as eva from '@eva-design/eva';
import React from 'react';
import store from '../../redux/store';
import { reduxForm } from 'redux-form';
import { Provider } from 'react-redux';
import { ApplicationProvider } from '@ui-kitten/components';
import TestRenderer from 'react-test-renderer';
import { OrderForm } from './Order';

const spy = jest.fn()
const Decorated = reduxForm({
  form: 'testForm', onSubmit: { spy }
})(OrderForm)

const formFieldValues = { name: 'email' }

it('should renders correctly', () => {
  const tree = TestRenderer.create(
    <ApplicationProvider {...eva} theme={eva.light}>
      <Provider store={store}>
        <Decorated 
          {...formFieldValues}
          orderDishes={['pizza']} totalPrice={100}
        />
      </Provider>
    </ApplicationProvider>

  ).toJSON()

  expect(tree).toMatchSnapshot()
})