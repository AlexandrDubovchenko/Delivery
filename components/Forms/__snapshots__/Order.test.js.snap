// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    category={true}
    style={
      Object {
        "flex": 3,
      }
    }
  >
    <View
      style={
        Object {
          "marginHorizontal": 15,
          "marginVertical": 5,
        }
      }
    >
      <View
        style={Object {}}
      >
        <View
          style={
            Array [
              Object {
                "backgroundColor": "#F7F9FC",
                "borderColor": "#E4E9F2",
                "borderRadius": 4,
                "borderWidth": 1,
                "minHeight": 40,
                "paddingHorizontal": 8,
                "paddingVertical": 7,
              },
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "width": "100%",
              },
            ]
          }
        >
          <TextInput
            allowFontScaling={true}
            appearance="default"
            editable={true}
            onBlur={[Function]}
            onChangeText={[Function]}
            onFocus={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            placeholder="Имя"
            placeholderTextColor="#FFD700"
            rejectResponderTermination={true}
            size="medium"
            status="basic"
            style={
              Array [
                Object {
                  "color": "#222B45",
                  "fontFamily": "System",
                  "fontSize": 15,
                  "fontWeight": "normal",
                  "marginHorizontal": 8,
                },
                Object {
                  "flexBasis": "auto",
                  "flexGrow": 1,
                  "flexShrink": 1,
                },
                null,
                Object {
                  "color": "black",
                },
              ]
            }
            underlineColorAndroid="transparent"
            value=""
          />
        </View>
        <View
          style={
            Array [
              Object {
                "marginTop": 4,
              },
              Object {
                "alignItems": "center",
                "flexDirection": "row",
              },
            ]
          }
        />
      </View>
    </View>
    <View>
      <View
        style={
          Object {
            "marginHorizontal": 15,
            "marginVertical": 5,
          }
        }
      >
        <View
          style={Object {}}
        >
          <View
            style={
              Array [
                Object {
                  "backgroundColor": "#F7F9FC",
                  "borderColor": "#E4E9F2",
                  "borderRadius": 4,
                  "borderWidth": 1,
                  "minHeight": 40,
                  "paddingHorizontal": 8,
                  "paddingVertical": 7,
                },
                Object {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "width": "100%",
                },
              ]
            }
          >
            <TextInput
              allowFontScaling={true}
              appearance="default"
              editable={true}
              onBlur={[Function]}
              onChangeText={[Function]}
              onFocus={[Function]}
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
              placeholder="Aдрес"
              placeholderTextColor="#FFD700"
              rejectResponderTermination={true}
              size="medium"
              status="basic"
              style={
                Array [
                  Object {
                    "color": "#222B45",
                    "fontFamily": "System",
                    "fontSize": 15,
                    "fontWeight": "normal",
                    "marginHorizontal": 8,
                  },
                  Object {
                    "flexBasis": "auto",
                    "flexGrow": 1,
                    "flexShrink": 1,
                  },
                  null,
                  Object {
                    "color": "black",
                  },
                ]
              }
              underlineColorAndroid="transparent"
              value=""
            />
          </View>
          <View
            style={
              Array [
                Object {
                  "marginTop": 4,
                },
                Object {
                  "alignItems": "center",
                  "flexDirection": "row",
                },
              ]
            }
          />
        </View>
      </View>
      <View
        style={
          Object {
            "flex": 1,
            "position": "absolute",
            "right": 30,
            "top": 7,
          }
        }
      >
        <RNGestureHandlerButton
          collapsable={false}
          onGestureEvent={[Function]}
          onGestureHandlerEvent={[Function]}
          onGestureHandlerStateChange={[Function]}
          onHandlerStateChange={[Function]}
          rippleColor={0}
          style={
            Array [
              Object {
                "overflow": "hidden",
              },
              undefined,
            ]
          }
        >
          <View
            accessible={true}
            style={
              Object {
                "opacity": 1,
              }
            }
          >
            <Image
              source={
                Object {
                  "testUri": "../../../assets/Orders/map_icon.png",
                }
              }
              style={
                Object {
                  "height": 40,
                  "width": 30,
                }
              }
            />
          </View>
        </RNGestureHandlerButton>
      </View>
    </View>
    <View
      style={
        Object {
          "marginHorizontal": 15,
          "marginVertical": 5,
        }
      }
    >
      <View
        style={Object {}}
      >
        <View
          style={
            Array [
              Object {
                "backgroundColor": "#F7F9FC",
                "borderColor": "#E4E9F2",
                "borderRadius": 4,
                "borderWidth": 1,
                "minHeight": 40,
                "paddingHorizontal": 8,
                "paddingVertical": 7,
              },
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "width": "100%",
              },
            ]
          }
        >
          <TextInput
            allowFontScaling={true}
            appearance="default"
            editable={true}
            onBlur={[Function]}
            onChangeText={[Function]}
            onFocus={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            placeholder="Телефон"
            placeholderTextColor="#FFD700"
            rejectResponderTermination={true}
            size="medium"
            status="basic"
            style={
              Array [
                Object {
                  "color": "#222B45",
                  "fontFamily": "System",
                  "fontSize": 15,
                  "fontWeight": "normal",
                  "marginHorizontal": 8,
                },
                Object {
                  "flexBasis": "auto",
                  "flexGrow": 1,
                  "flexShrink": 1,
                },
                null,
                Object {
                  "color": "black",
                },
              ]
            }
            underlineColorAndroid="transparent"
            value=""
          />
        </View>
        <View
          style={
            Array [
              Object {
                "marginTop": 4,
              },
              Object {
                "alignItems": "center",
                "flexDirection": "row",
              },
            ]
          }
        />
      </View>
    </View>
    <Text
      style={
        Object {
          "fontSize": 20,
          "marginBottom": 30,
          "marginHorizontal": 15,
          "marginTop": 10,
        }
      }
    >
      Сумма заказа:
      100
      грн
    </Text>
    <RNGestureHandlerButton
      collapsable={false}
      onGestureEvent={[Function]}
      onGestureHandlerEvent={[Function]}
      onGestureHandlerStateChange={[Function]}
      onHandlerStateChange={[Function]}
      rippleColor={0}
      style={
        Array [
          Object {
            "overflow": "hidden",
          },
          undefined,
        ]
      }
    >
      <View
        accessible={true}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#FFD700",
            "marginHorizontal": 15,
            "opacity": 1,
          }
        }
      >
        <Text
          appearance="default"
          category="h5"
          style={
            Array [
              Object {
                "color": "#222B45",
                "fontFamily": "System",
                "fontSize": 22,
                "fontWeight": "800",
              },
              undefined,
            ]
          }
        >
          Отправить заказ
        </Text>
      </View>
    </RNGestureHandlerButton>
  </View>
</View>
`;
